pkg/svcat/service-catalog/broker_test.go

```diff
@@ -0,0 +1,135 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/testing"
+
+	. "github.com/kubernetes-incubator/service-catalog/pkg/svcat/service-catalog"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Broker", func() {
+	var (
+		sdk          *SDK
+		svcCatClient *fake.Clientset
+		sb           *v1beta1.ClusterServiceBroker
+		sb2          *v1beta1.ClusterServiceBroker
+	)
+
+	BeforeEach(func() {
+		sb = &v1beta1.ClusterServiceBroker{ObjectMeta: metav1.ObjectMeta{Name: "foobar"}}
+		sb2 = &v1beta1.ClusterServiceBroker{ObjectMeta: metav1.ObjectMeta{Name: "barbaz"}}
+		svcCatClient = fake.NewSimpleClientset(sb, sb2)
+		sdk = &SDK{
+			ServiceCatalogClient: svcCatClient,
+		}
+	})
+
+	Describe("RetrieveBrokers", func() {
+		It("Calls the generated v1beta1 List method", func() {
+			brokers, err := sdk.RetrieveBrokers()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(brokers).Should(ConsistOf(*sb, *sb2))
+			Expect(svcCatClient.Actions()[0].Matches("list", "clusterservicebrokers")).To(BeTrue())
+		})
+		It("Bubbles up errors", func() {
+			badClient := &fake.Clientset{}
+			errorMessage := "error retrieving list"
+			badClient.AddReactor("list", "clusterservicebrokers", func(action testing.Action) (bool, runtime.Object, error) {
+				return true, nil, fmt.Errorf(errorMessage)
+			})
+			sdk.ServiceCatalogClient = badClient
+			_, err := sdk.RetrieveBrokers()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring(errorMessage))
+			Expect(badClient.Actions()[0].Matches("list", "clusterservicebrokers")).To(BeTrue())
+		})
+	})
+	Describe("RetrieveBroker", func() {
+		It("Calls the generated v1beta1 List method with the passed in broker", func() {
+			broker, err := sdk.RetrieveBroker(sb.Name)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(broker.Name).To(Equal(sb.Name))
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "clusterservicebrokers")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(sb.Name))
+		})
+		It("Bubbles up errors", func() {
+			brokerName := "banana"
+
+			broker, err := sdk.RetrieveBroker(brokerName)
+
+			Expect(broker).To(BeNil())
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("not found"))
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "clusterservicebrokers")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(brokerName))
+		})
+	})
+	Describe("RetrieveBrokerByClass", func() {
+		It("Calls the generated v1beta1 List method with the passed in class's parent broker", func() {
+			sc := &v1beta1.ClusterServiceClass{Spec: v1beta1.ClusterServiceClassSpec{ClusterServiceBrokerName: sb.Name}}
+			broker, err := sdk.RetrieveBrokerByClass(sc)
+
+			Expect(broker).NotTo(BeNil())
+			Expect(err).NotTo(HaveOccurred())
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "clusterservicebrokers")).To(BeTrue())
```

Comment:
Same question as before. Just wondering if we can also verify that the filter condition was applied correctly?

Reply:
[https://github.com/kubernetes-incubator/service-catalog/blob/master/vendor/k8s.io/client-go/testing/actions.go](The generated fakes I'm using) are pretty limited in what they give you access to. I can't figure out a way to get that info out of them.

Reply:
https://github.com/kubernetes-incubator/service-catalog/blob/3032f01c8a04270c27d30add13dc4d4d37a32cad/vendor/k8s.io/client-go/testing/actions.go#L78

That gives me hope but is totally something that could be looking into later.